jib-core/src/main/java/com/google/cloud/tools/jib/cache/DefaultCacheStorageWriter.java

```diff
@@ -101,6 +110,36 @@ private static DescriptorDigest getDiffIdByDecompressingFile(Path compressedFile
     }
   }
 
+  /**
+   * Writes a json template to the destination path by writing to a temporary file then moving the
+   * file.
+   *
+   * @param jsonTemplate the json template
+   * @param destination the destination path
+   * @throws IOException if an I/O exception occurs
+   */
+  private static void writeMetadata(JsonTemplate jsonTemplate, Path destination)
+      throws IOException {
+    try (LockFile ignored1 = LockFile.lock(Paths.get(destination.toString() + ".lock"))) {
```

Comment:
Just curious: does `FileLock` support read/write locks, or is it always mutual exclusion? (Not that it matters even if mutual exclusion for reading.)